pkg/primaza/controlplane/push_to_service_namespaces.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2023 The Primaza Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controlplane
+
+import (
+	"context"
+
+	primazaiov1alpha1 "github.com/primaza/primaza/api/v1alpha1"
+	"github.com/primaza/primaza/pkg/primaza/clustercontext"
+	"github.com/primaza/primaza/pkg/slices"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func PushServiceClassToServiceNamespaces(ctx context.Context, sc primazaiov1alpha1.ServiceClass, ce primazaiov1alpha1.ClusterEnvironment, scheme *runtime.Scheme, mapper meta.RESTMapper, controllerruntimeClient client.Client, failedServiceNamespaces []string) error {
+	cfg, err := clustercontext.GetClusterRESTConfig(ctx, controllerruntimeClient, ce.Namespace, ce.Spec.ClusterContextSecret)
+	if err != nil {
+		return err
+	}
+
+	oc := client.Options{
+		Scheme: scheme,
+		Mapper: mapper,
+	}
+	cli, err := client.New(cfg, oc)
+	if err != nil {
+		return err
+	}
+	sns := slices.SubtractStr(ce.Spec.ServiceNamespaces, failedServiceNamespaces)
```

Comment:
Have made the change to not pass the clusterEnvironment and failedServiceNamespaces parameters. Not much value added by this though because we still needed to pass another param for rest.Config.